src/Data/LLVM/BitCode/IR/Values.hs

```diff
@@ -44,15 +44,20 @@ getValueTypePair t r ix = do
 --getValueNoFwdRef :: Type -> Int -> Parse (Typed PValue)
 --getValueNoFwdRef ty n = label "getValueNoFwdRef" (getFnValueById ty =<< adjustId n)
 
-getFnValueById :: Type -> Int -> Parse (Typed PValue)
+getFnValueById :: (HasMdTable m, HasParseEnv m, HasValueTable m, MonadFail m)
+               => Type -> Int -> m (Typed PValue)
 getFnValueById  = getFnValueById' Nothing
 
 getValue :: ValueTable -> Type -> Int -> Parse (Typed PValue)
 getValue vt ty n = label "getValue" (getFnValueById' (Just vt) ty =<< adjustId n)
 
 -- | Lookup a value by its absolute id, or perhaps some metadata.
-getFnValueById' :: Maybe ValueTable -> Type -> Int -> Parse (Typed PValue)
-getFnValueById' mbVt ty n = label "getFnValueById'" $ case ty of
+getFnValueById' :: HasMdTable m
+                => HasParseEnv m
+                => HasValueTable m
+                => MonadFail m
+                => Maybe ValueTable -> Type -> Int -> m (Typed PValue)
```

Comment:
The type signature for `getFnValueById` puts all of the constraints into a tuple:

```hs
getFnValueById :: (HasMdTable m, HasParseEnv m, HasValueTable m, MonadFail m)
               => ...
```

While the type signature for `getFnValueById'` puts each constraint on its own line:

```hs
getFnValueById' :: HasMdTable m
                => HasParseEnv m
                => HasValueTable m
                => MonadFail m
                => ...
```

It would be better to stick to a single convention, I think. I have a slight preference towards the former style, although I wouldn't push back if you advocate for the latter style.

Reply:
I have a slight preference for the stacked 1/line style because it makes commenting out individual constraints during dev easier.  That said, it's only a very slight preference, and I'm fine with tupling... so: done.